test/sorting.test.tsx

```diff
@@ -0,0 +1,107 @@
+import userEvent from '@testing-library/user-event';
+import { getHeaderCells } from './utils';
+import type { Column, SortColumn } from '../src/types';
+import { StrictMode, useState } from 'react';
+import DataGrid from '../src';
+import { render, screen } from '@testing-library/react';
+
+const columns: readonly Column<SortColumn>[] = [
+  { key: 'colA', name: 'colA' },
+  { key: 'colB', name: 'colB' },
+  { key: 'colC', name: 'colC' }
+];
+
+function TestGrid() {
+  const [sortColumns, setSortColumns] = useState<readonly SortColumn[]>([]);
+
+  return (
+    <>
+      <DataGrid
+        columns={columns}
+        rows={[]}
+        defaultColumnOptions={{ sortable: true }}
+        sortColumns={sortColumns}
+        onSortColumnsChange={setSortColumns}
+      />
+      <div data-testid="sortColumnsValue">{JSON.stringify(sortColumns)}</div>
+    </>
+  );
+}
+
+function setup() {
+  return render(
+    <StrictMode>
+      <TestGrid />
+    </StrictMode>
+  );
+}
+
+function testSortColumns(expectedValue: readonly SortColumn[]) {
+  expect(JSON.parse(screen.getByTestId('sortColumnsValue')!.textContent!)).toStrictEqual(
```

Comment:
Might be missing some eslint rules here
```suggestion
  expect(JSON.parse(screen.getByTestId('sortColumnsValue').textContent!)).toStrictEqual(
```